packages/transform/src/template/inlining/index.ts

```diff
@@ -10,49 +10,21 @@ export type CorrelatedSpansResult = {
   partialSpans: Array<PartialCorrelatedSpan>;
 };
 
-export type ContainingTypeInfo = {
-  inClass: boolean;
-  className: string | undefined;
-  contextType: string | undefined;
-  typeParams: string | undefined;
-};
-
 /**
- * Given an AST node for an embedded template, determines the appropriate
- * instance type to be passed to `@glint/template`'s `ResolveContext`, as well
- * as any type parameters that need to be propagated as inputs to the template's
- * root generator function.
- *
- * For example, a template declared within `class MyComponent<T extends string>`
- * would give `MyComponent<T>` as the context type, and `<T extends string>` as
- * the type params, ultimately resulting in a template function like:
- *
- *     template(function*<T extends string>(𝚪: ResolveContext<MyComponent<T>>){
- *       // ...
- *     })
+ * Given an AST node for an embedded template, determines whether it's embedded
+ * within a class in such a way that that class should be treated as its backing
+ * value.
  */
-export function getContainingTypeInfo(ts: TSLib, node: ts.Node): ContainingTypeInfo {
-  let container = findContainingClass(ts, node);
-  let inClass = Boolean(container);
-  let className = container?.name?.text;
-  let contextType = className;
-  let typeParams = undefined;
-
-  if (container?.typeParameters) {
-    let params = container.typeParameters;
-    typeParams = `<${params.map((param) => param.getText()).join(', ')}>`;
-    contextType += `<${params.map((param) => param.name.getText()).join(', ')}>`;
-  }
-
-  return { contextType, typeParams, className, inClass };
-}
-
-function findContainingClass(ts: TSLib, node: ts.Node): ts.ClassLikeDeclaration | null {
+export function isEmbeddedInClass(ts: TSLib, node: ts.Node): boolean {
   let current: ts.Node | null = node;
   do {
+    // TODO: this should likely actually filter on whether the template appears in a
+    // static block or property definition, but just "am I in a class body" is the
+    // current status quo and has been ok so far.
```

Comment:
What would be the difference between those two from *our* POV? (Obviously it matters from a runtime and performance POV, but it's not obvious to me why we would choose to care about it.)

Reply:
If I wrote something like this today:

```tsx
class MyComponent {
  InternalThing = <template>Hello!</template>;

  <template>
    <this.InternalThing />
  </template>
}
```

Then the template-only component we're assigning to `InternalThing` would be treated as a `templateForBackingClass`, which is incorrect.

Reply:
It's the status quo and no one has run into it yet to my knowledge, which is why I didn't want to go down the rabbithole of actually fixing it as part of this and further delay releasing, but I _did_ want to leave a note that the behavior's incorrect

Reply:
Ahhhh, that example is clarifying. Thank you.